packages/apps/comment-nuke/src/nuke.ts

```diff
@@ -1,13 +1,128 @@
-import { Comment, Devvit } from '@devvit/public-api';
-
+import { Comment, Devvit, Post } from '@devvit/public-api';
+ 
 export type NukeProps = {
   remove: boolean;
   lock: boolean;
   skipDistinguished: boolean;
   commentId: string;
   subredditId: string;
 };
-
+ 
+export type NukePostProps = {
+    remove: boolean;
+    lock: boolean;
+    skipDistinguished: boolean;
+    postId: string;
+    subredditId: string;
+  };
+ 
+   async function getAllCommentsInPost(
+    post: Post,
+    result: Comment[] = []
+  ): Promise<Comment[]> {
+    const comments = await post.comments.all();
+    if (comments.length > 0) {
+      for (const comment of comments) {
+        await getAllCommentsInThread(comment, result);
+      }
+    }
+    return result;
+  }
+  
+  export async function handleNukePost(props: NukePostProps, context: Devvit.Context) {
+    const startTime = new Date();
+    let success = true;
+    let message: string;
+  
+    const shouldLock = props.lock;
+    const shouldRemove = props.remove;
+    const skipDistinguished = props.skipDistinguished;
+  
+    if (!shouldLock && !shouldRemove) {
+      return { message: 'You must select either lock or remove.', success: false };
+    }
+  
+    try {
+      const post = await context.reddit.getPostById(props.postId);
+      const user = await context.reddit.getCurrentUser();
+      const modPermissions = await user.getModPermissionsForSubreddit(post.subredditName);
```

Comment:
I think we need a null check on user first. something like:

if (!user) return {success: false, message: "can't get user"}

the typechecker should flag this.